### PR TITLE
ci: bump go to 1.15.5 and a few deps, use new quay images

### PR DESCRIPTION
**Pull request by kolyshkin**
#### What type of PR is this?

/kind ci
/kind dependency-change

##### 1. ci: bump go to 1.15.5

go1.15.4 (released 2020/11/05) includes fixes to cgo, the compiler,
linker, runtime, and the compress/flate, net/http, reflect, and time
packages. See the [Go 1.15.4 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.15.4+label%3ACherryPickApproved) for details.

go1.15.5 (released 2020/11/12) includes security fixes to the cmd/go
and math/big packages. See the [Go 1.15.5 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.15.5+label%3ACherryPickApproved) for details.

##### 2. contrib/test/int/build: bump a few deps
    
    conmon:      v2.0.15 -> v2.0.20
    bats:        v1.2.0  -> v1.2.1
    cni-plugins: v0.8.4  -> v0.8.7
    
Those were already bumped for circleci multiple times
(see commits 65fe2c5, d9ea392, 40b9d971fe, e48d23aab)
but not here.

(this used to be in #4383)

##### 3. circleci: use updated images
    
Images are rebuilt with latest golang version and updated dependencies.
Used master-1.6.1 tag since master-1.6.0 is somehow already taken and
I did not want to remove it.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ executors:
 
   container-base:
     docker:
-      - image: golang:1.14
+      - image: golang:1.15
```
Comment (saschagrunert) [inline .circleci/config.yml:35]: Do we also want to rebuild the CI images? It should pick the latest go version automatically but we have to bump/push a new version.

File: scripts/versions [hunk 2]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 # Versions to be used
 declare -A VERSIONS=(
-    ["go"]=1.15.3
+    ["go"]=1.15.4
```
Comment (haircommander) [inline scripts/versions:6]: going along with what @saschagrunert said above, after adding this, one needs to run `make test-images` and push the images to quay and update the image locations in the `stdenv` section of .circleci/config.yml

Comment (kolyshkin) [reply scripts/versions:6]: Alas, I don't know much about it... Can you please do that?

I mean, update from 1.15.3 from 1.15.4 is probably not that important; but 1.14 to 1.15 might be.

File: .circleci/config.yml (hunk 1 above)
Comment (kolyshkin) [reply .circleci/config.yml:35]: done

File: scripts/versions (hunk 2 above)
Comment (kolyshkin) [reply scripts/versions]: done

**Comment by kolyshkin** [comment]
CI failure in integration-static is a flake:

> not ok 2 pids limit
> (in test file ./cgroups.bats, line 22)
>   `ctr_id=$(crictl run "$TESTDIR"/container_pids_limit.json "$TESTDATA"/sandbox_config.json)' failed
> ...
> time="2020-11-18T05:02:01Z" level=fatal msg="running container: creating container failed: rpc error: code = Unknown desc = Error reading blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4: Get \"https://cdn02.quay.io/sharedimages/86f0a285-6f29-47c4-a3ae-7e2c70cad0ba/layer?Expires=1605676304&Signature=N976UlLi1Hb~jhKGQAy8JcmhlqFPdTzI3I3WiR7iEmb-x1YaIA6cWgWB3DffhwlaOrkzwHU8bAuWUEcwbD1N4tx3tAy7rhfQSyqWzbrs-OZJGON9aH7C5PkoqYPFDwGxQ1dzxtLSFR-hx1FmyN1lIuzFUKRmiQlPZPj14VGh51olcYgw2QVOPHE7~asGYa7EtQO~9q9A4Cgo7AAKzVsgDTjw19KReyIl5yh8am272W0AMrCoELwqC83F0FV3i6IZsomhnm30BHa4D5Xz1Jzr-dy4VWpM-Y8mowR4AVH9PLi6aALmYk1Ei54QjDEtkjSwc4sIVISb7g2uiKSDbLFHbQ__&Key-Pair-Id=APKAJ67PQLWGCSP66DGA\": net/http: TLS handshake timeout"

Opened https://github.com/cri-o/cri-o/pull/4377

**Comment by kolyshkin** [comment]
rebase to include #4378

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by kolyshkin** [comment]
e2e_crun failure is a flake addressed by https://github.com/cri-o/cri-o/pull/4385

/test e2e_crun

**Comment by kolyshkin** [comment]
* merged #4383 into this one
* rebased on top of current HEAD
* rebuilt and uploaded the images

**Comment by fidencio** [comment]
/lgtm
Thanks, @kolyshkin

**Pull request merged**